modelci/hub/model_loader.py

```diff
@@ -38,22 +39,21 @@ def savedmodel_loader(model_weight_path: Path):
     return tf.keras.models.load_model(model_weight_path)
 
 
-def load(model_path: os.PathLike):
+def load(model: MLModel):
+    # TODO only support torch.save, saved_model, and joblib serialization for the time being
     """A unify API to load model weight files in various format.
 
     Args:
-        model_path: Path to the entire model file. The model is saved in ModelCI standard directory.
+        model: MLModel
     """
-
-    model_path = Path(model_path)
-    try:
-        model_info = parse_path_plain(model_path)
-    except ValueError as e:
-        # TODO: handle other path format, e.g. torch hub
-        raise e
-    if model_info['framework'] == 'PYTORCH' and model_info['engine'] in ('NONE', 'PYTORCH'):  # PyTorch
-        return pytorch_loader(model_path)
-    elif model_info['framework'] == 'TENSORFLOW' and model_info['engine'] in ('NONE', 'TENSORFLOW'):  # TensorFlow
-        return savedmodel_loader(model_path)
-    elif model_info['framework'] in ('SKLEARN', 'XGBOOST', 'LIGHTGBM') and model_info['engine'] == 'NONE':  # sklearn
-        return joblib_loader(model_path)
+    if not Path(model.saved_path).exists():
+        (filepath, filename) = os.path.split(model.saved_path)
+        os.makedirs(filepath)
+        with open(model.saved_path, 'wb') as f:
+            f.write(model.weight.__bytes__())
```

Comment:
It is very strange to modify the weight in a `load` function, you should re-consider your changes to it.

Reply:
The `load` function receives a `MLModel` and returns a `torch.nn.Module` or other model forms corresponding to the serialization way of the model.  The `MLModel `keeps a binary object of serialized model. To load this model, `load` function firstly go to `MLModel.saved_path` to find if there is a cache file in local file system. If there is no cache file, then the `load` function will generate a cache file from `MLModel`'s binary object at this path, and finally load the model. So, it is more a cache file processing behavior than  a modification to the model weight.

Reply:
My suggestion is to use cache manager to check and download the cache in whichever files before invoking the model loader, rather than do it manually here.

Reply:
Thanks, I will make this change.